Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalSiliconRotatedModule.cc

```diff
@@ -0,0 +1,462 @@
+      glog.placeVolume(ns.volume(wafer), copy, tran);
+#ifdef EDM_ML_DEBUG
+      ++ntype[type];
+      edm::LogVerbatim("HGCalGeom") << " DDHGCalSiliconRotatedModule: " << wafer << " number " << copy << " type "
+                                    << layertype << ":" << type << " positioned in " << glog.name() << " at ("
+                                    << cms::convert2mm(xpos) << "," << cms::convert2mm(ypos) << ",0) with no rotation";
+#endif
+    }
+
+#ifdef EDM_ML_DEBUG
+    edm::LogVerbatim("HGCalGeom") << "DDHGCalSiliconRotatedModule: Maximum # of u " << ium << " # of v " << ivm
+                                  << " and " << kount << " wafers (" << ntype[0] << ":" << ntype[1] << ":" << ntype[2]
+                                  << ") for " << glog.name();
+#endif
+  }
+
+  //Required data members to cache the values from XML file
+  HGCalGeomTools geomTools_;
+  HGCalCassette cassette_;
+
+  int waferTypes_;                      // Number of wafer types
+  int facingTypes_;                     // Types of facings of modules toward IP
+  int orientationTypes_;                // Number of wafer orienations
+  int placeOffset_;                     // Offset for placement
+  int firstLayer_;                      // Copy # of the first sensitive layer
+  int absorbMode_;                      // Absorber mode
+  int sensitiveMode_;                   // Sensitive mode
+  double zMinBlock_;                    // Starting z-value of the block
+  double waferSize_;                    // Width of the wafer
+  double waferSepar_;                   // Sensor separation
+  int sectors_;                         // Sectors
+  int cassettes_;                       // Cassettes
+  std::string rotstr_;                  // Rotation matrix (if needed)
+  std::vector<std::string> waferFull_;  // Names of full wafer modules
+  std::vector<std::string> waferPart_;  // Names of partial wafer modules
+  std::vector<std::string> materials_;  // names of materials
+  std::vector<std::string> names_;      // Names of volumes
+  std::vector<double> thick_;           // Thickness of the material
+  std::vector<int> copyNumber_;         // Initial copy numbers
+  std::vector<int> layers_;             // Number of layers in a section
+  std::vector<double> layerThick_;      // Thickness of each section
+  std::vector<int> layerType_;          // Type of the layer
+  std::vector<int> layerSense_;         // Content of a layer (sensitive?)
+  std::vector<double> slopeB_;          // Slope at the lower R
+  std::vector<double> zFrontB_;         // Starting Z values for the slopes
+  std::vector<double> rMinFront_;       // Corresponding rMin's
+  std::vector<double> slopeT_;          // Slopes at the larger R
+  std::vector<double> zFrontT_;         // Starting Z values for the slopes
+  std::vector<double> rMaxFront_;       // Corresponding rMax's
+  std::vector<int> layerTypes_;         // Layer type (Centering, rotations..)
```

Comment:
Since it is not obvious to me to understand it from the code: are really `layerTipe_` and `layerTypes_` different?
Any less ambiguous name for (at least) one of the two could not be invented?

Reply:
They are indeed different - one rferes to all the layers and the other to the layer containing the wafer as described in the lfat file. May be we shall change one of the names in the next revision. Please merge this PR as it is.